Document binary sensors for Shelly inputs (#20047)
home-assistant/home-assistant.github.io: source/_integrations/shelly.markdown

- If `Channel Name` is set in the device, the integration will use it to generate the entities' name, e.g. `Kitchen Light`
- If `Channel Name` is set to the default value, the integration will use the `Device ID` and default channel name to generate the entities' name, e.g. `ShellyPro4PM-9808D1D8B912 switch_0`.

## Binary input sensors

> Depending on how a device's button type is configured, the integration will create binary sensors corresponding to those inputs. Starting with version 2021.10.4 of Home Assistant, these binary sensors will not be created when the button type is `momentary` or `momentary_on_release`.

Reviewer: We don't keep a changelog in our documentation. Instead, we document only the latest state.

That said, 2021.10.4 is a patch release, which should not contain breaking changes. If it does, it is more likely a bug that should be reported.

Reviewer 2: That comment is wrong. Binary sensors were never created when button type is `momentary` or `momentary_on_release`.

Reviewer 2: ```suggestion
## Binary input sensors (generation 1)
Depending on how a device's button type is configured, the integration will create binary sensors corresponding to those inputs. binary sensors are not created when the button type is `momentary` or `momentary_on_release`, for these types you need to use events for your automations.
```

> 
> For generation 2 hardware it's possible to select if a device's input is connected to a button or a switch. Binary sensors are created only if the input mode is set to `switch`. When the input is of type `button` you need to use events for your automations.

Reviewer 2: ```suggestion

## Binary input sensors (generation 2)
For generation 2 hardware it's possible to select if a device's input is connected to a button or a switch. Binary sensors are created only if the input mode is set to `switch`. When the input is of type `button` you need to use events for your automations.
```

- If `Channel Name` is set in the device, the integration will use it to generate the entities' name, e.g. `Kitchen Light`
- If `Channel Name` is set to the default value, the integration will use the `Device ID` and default channel name to generate the entities' name, e.g. `ShellyPro4PM-9808D1D8B912 switch_0`.

> ## Binary input sensors
> ## Binary input sensors (generation 1)
> Depending on how a device's button type is configured, the integration will create binary sensors corresponding to those inputs. binary sensors are not created when the button type is `momentary` or `momentary_on_release`, for these types you need to use events for your automations.
> 
> ## Binary input sensors (generation 2)
> For generation 2 hardware it's possible to select if a device's input is connected to a button or a switch. Binary sensors are created only if the input mode is set to `switch`. When the input is of type `button` you need to use events for your automations.

Reviewer: ```suggestion
## Binary input sensors

### Binary input sensors (generation 1)

Depending on how a device's button type is configured, the integration will create binary sensors corresponding to those inputs. binary sensors are not created when the button type is `momentary` or `momentary_on_release`, for these types you need to use events for your automations.

### Binary input sensors (generation 2)

For generation 2 hardware it's possible to select if a device's input is connected to a button or a switch. Binary sensors are created only if the input mode is set to `switch`. When the input is of type `button` you need to use events for your automations.
```
